ors-api/src/main/java/org/heigit/ors/api/controllers/SnappingAPI.java

```diff
@@ -0,0 +1,157 @@
+@ApiResponse(responseCode = "501", description = "Indicates that the server does not support the functionality needed to fulfill the request.")
+@ApiResponse(responseCode = "503", description = "The server is currently unavailable due to overload or maintenance.")
+public class SnappingAPI {
+    static final CommonResponseEntityExceptionHandler errorHandler = new CommonResponseEntityExceptionHandler(SnappingErrorCodes.BASE);
+
+    private final SnappingService snappingService;
+
+    public SnappingAPI(SnappingService snappingService) {
+        this.snappingService = snappingService;
+    }
+
+    // generic catch methods - when extra info is provided in the url, the other methods are accessed.
+    @GetMapping
+    @Operation(hidden = true)
+    public void getGetMapping() throws MissingParameterException {
+        throw new MissingParameterException(SnappingErrorCodes.MISSING_PARAMETER, "profile");
+    }
+
+    @PostMapping
+    @Operation(hidden = true)
+    public String getPostMapping(@RequestBody SnappingApiRequest request) throws MissingParameterException {
+        throw new MissingParameterException(SnappingErrorCodes.MISSING_PARAMETER, "profile");
+    }
+
+    // Matches any response type that has not been defined
+    @PostMapping(value = "/{profile}/*")
+    @Operation(hidden = true)
+    public void getInvalidResponseType() throws StatusCodeException {
+        throw new StatusCodeException(HttpServletResponse.SC_NOT_ACCEPTABLE, SnappingErrorCodes.UNSUPPORTED_EXPORT_FORMAT, "This response format is not supported");
+    }
+
+    // Functional request methods
+    @PostMapping(value = "/{profile}")
+    @Operation(
+            description = """
+                    Returns a list of points snapped to the nearest edge in the graph. \
+                    This method does not accept any request body or parameters other than profile, locations, maximum_snapping_radius.\
+                    """,
+            summary = "Snapping Service"
+    )
+    @ApiResponse(
+            responseCode = "200",
+            description = "Standard response for successfully processed requests. Returns JSON.",
+            content = {@Content(
+                    mediaType = "application/geo+json",
+                    schema = @Schema(implementation = JsonSnappingResponse.class)
+            )
+            })
+    public JsonSnappingResponse getDefault(@Parameter(description = "Specifies the route profile.", required = true, example = "driving-car") @PathVariable APIEnums.Profile profile,
+                                         @Parameter(description = "The request payload", required = true) @RequestBody SnappingApiRequest request) throws StatusCodeException {
+        return getJsonSnapping(profile, request);
+    }
+
+    @PostMapping(value = "/{profile}/json", produces = {"application/json;charset=UTF-8"})
+    @Operation(
+            description = "Returns a list of points snapped to the nearest edge in the graph.",
```

Comment:
This should probably also contain the description of the other snapping endpoint, as they do the same.

Is there a reason for having always the `/json` endpoint as well?

Reply:
Done.

Reply:
the `/json` is there as we have the same structure in the other apis as well. Once we have `/geojson` it makes more sense again as well.